dm/config/task.go

```diff
@@ -775,11 +777,11 @@ func (c *TaskConfig) adjust() error {
 		instanceIDs[inst.SourceID] = i
 
 		switch c.TaskMode {
-		case ModeFull, ModeAll:
+		case ModeFull, ModeAll, ModeDump:
 			if inst.Meta != nil {
 				log.L().Warn("metadata will not be used. for Full mode, incremental sync will never occur; for All mode, the meta dumped by MyDumper will be used", zap.Int("mysql instance", i), zap.String("task mode", c.TaskMode))
 			}
-		case ModeIncrement:
+		case ModeIncrement, ModeLoadSync:
```

Comment:
do we require the data source files of load unit having a metadata file? or what if the metadata file and task metadata both have binlog starting position?

Reply:
We require the data source files of load unit having a metadata file. If the metadata file and task metadata both have binlog starting position we should use the data in metadata file.
BTW, these two task modes will only be used in cloud in our design. They are not planned to be exposed to other users.